js/app.js

```diff
@@ -200,61 +201,83 @@ label_list.on('contextmenu', function (e) {
 });
 
 //edit label name
-label_list.on('dblclick', '.label', function () {
+$('#label-list').on('dblclick', '.label', function () {
   //enble editing
   $(this).children('.label-name')[0].contentEditable = true;
   //open textbox
   $(this).children('.label-name').focus().selectText();
 });
 
 // user pressed enter on label name change
-label_list.on('keypress', '.label-name', function (e) {
+$('#label-list').on('keypress', '.label-name', function (e) {
   if (e.which === 13) {
     $(this).blur();
   }
 });
 
 //stopped editing label name
-label_list.on('blur', '.label-name', function () {
+$('#label-list').on('blur', '.label-name', function () {
   //disable editing
   this.contentEditable = false;
 
   //fix whitespace and create new label name with no spaces (class names can't have spaces)
-  let newName = $(this).text().trim().replace(/\s+/g, "_").replace(/<|>|&/g, '');
-  $(this).text(newName);
-  console.log("Attempting to change label name from " + tagModel.currentCategory + " to " + newName);
+  let newLabelName = $(this).text().trim().replace(/\s+/g, "_").replace(/<|>|&/g, '');
+  $(this).text(newLabelName);
 
   //check if the name is the same as previous
-  if (newName === tagModel.currentCategory) {
+  if (newLabelName === tagModel.currentCategory) {
     console.log('Aborting: Category is the same name as before');
     return;
   }
 
-  //check for valid label name
-  if ((tagModel.categoryIndex(newName) >= 0) || newName === '') {
-    console.log('Aborting: Invalid label name: "' + newName + '"');
-    $(this).text(tagModel.currentCategory);
+  //check if user clicked off menu without first giving a label name
+  if (newLabelName === "") {
+    console.log('Aborting: Category name was not specified');
+    deleteLabel();
     return;
   }
 
+  //if this label already exists
+  if ((tagModel.categoryIndex(newLabelName) >= 0)) {
+    //and if there's no open document, then just delete the label
+    if (tagModel.currentDoc === null) {
+      console.log('Aborting label name change: already exists: "' + newLabelName + '"');
+      alert("This label already exists! Please try again.");
+      deleteLabel();
+      return;
+    }
+    //if there is a document, check for existing annotations
+    else {
+      var hasAnnotations = tagModel.currentDoc.checkIfLabelHasAnnotations(tagModel.currentCategory);
+
+      //if it has annotations, then restore the old label name
+      //if it doesnt, then we can just delete the html div and have the user start over
+      hasAnnotations ? $(this).text(tagModel.currentCategory) : deleteLabel();
```

Comment:
You fixed part of it, the error's gone on the same doc, but if you annotate a doc, and then go to a new doc with no annotations, and try to change the label, it throws the old error.  I'm guessing it only checks if there are annotations on the current document, and not the whole thing, and deleting the label references something not there, but eff me if i know.